sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,35 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   * - BROADCAST/BROADCASTJOIN/MAPJOIN match the closest table with the given name parameters.
```

Comment:
Yes. `Batch` is the guide to show that kind of dependecies. 

You know, every optimizer rules already have those dependency. And, `Batch` is the only way to show those dependencies we've got so far.